packages/block-library/src/post-title/edit.js

```diff
@@ -30,7 +30,19 @@ export default function PostTitleEdit( {
 } ) {
 	const TagName = 0 === level ? 'p' : 'h' + level;
 	const isDescendentOfQueryLoop = Number.isFinite( queryId );
-	const userCanEdit = useCanEditEntity( 'postType', postType, postId );
+	/**
+	 * Hack: useCanEditEntity may trigger an OPTIONS request. When the post title
+	 * is a descendant of a query loop, we know for sure that user cannot edit
+	 * the title. To avoid unnecessary OPTIONS requests we should avoid using
+	 * useCanEditEntity. However, it's a hook and cannot be fired conditionally,
+	 * hence instead we call it without proper data. In this case hook automatically
+	 * returns false without calling a backend.
+	 */
```

Comment:
Maybe some rewording like:

`useCanEditEntity` may trigger an OPTIONS request to the REST API via the `canUser` resolver. However, when the Post Title is a descendant of a Query Loop block, the title cannot be edited. In order to avoid these unnecessary requests, we call the hook without the proper data, resulting in returning early without making them.

Reply:
I like that, changed applied in [commit](https://github.com/WordPress/gutenberg/pull/49839/commits/048972ef20e03a07fc387d8944449cceb3e33007) 👍